### PR TITLE
[libc++] mark P0154 as implemented; fix column alignment issues

### PR DESCRIPTION
**Pull request by h-vetinari**
AFAIU, P0154 support is complete as of https://github.com/llvm/llvm-project/pull/89446

### CODE REVIEW

File: libcxx/docs/Status/Cxx17.rst [hunk 1]
```diff
@@ -42,7 +42,6 @@ Paper Status
 
    .. [#note-P0067] P0067: ``std::(to|from)_chars`` for integrals has been available since version 7.0. ``std::to_chars`` for ``float`` and ``double`` since version 14.0 ``std::to_chars`` for ``long double`` uses the implementation for ``double``.
    .. [#note-P0607] P0607: The parts of P0607 that are not done are the ``<regex>`` bits.
-   .. [#note-P0154] P0154: This is currently not implemented for clang because clang does not implement the required macros.
```
Comment (mordante) [inline libcxx/docs/Status/Cxx17.rst:45]: When LLVM-19 is released libc++ still supports Clang 18, which does not have the needed support. Therefore I prefer to keep the note and mention it requires Clang 19 or newer.

(I've no objection to mark the feature as complete.)

**Review by mordante — APPROVED** [review]
Thanks for catching this! LGTM modulo one comment.

File: libcxx/docs/Status/Cxx17.rst (hunk 1 above)
Comment (h-vetinari) [reply libcxx/docs/Status/Cxx17.rst:45]: Done

**Comment by mordante** [comment]
Do you have commit access?

**Comment by h-vetinari** [comment]
> Do you have commit access?

I do not... 🙃

File: libcxx/docs/Status/Cxx17Papers.csv [hunk 2]
```diff
@@ -35,7 +35,7 @@
 "`P0185R1 <https://wg21.link/P0185R1>`__","LWG","Adding [nothrow-]swappable traits","Jacksonville","|Complete|","3.9"
 "`P0253R1 <https://wg21.link/P0253R1>`__","LWG","Fixing a design mistake in the searchers interface","Jacksonville","|Complete|","3.9"
 "`P0025R0 <https://wg21.link/P0025R0>`__","LWG","An algorithm to ""clamp"" a value between a pair of boundary values","Jacksonville","|Complete|","3.9"
-"`P0154R1 <https://wg21.link/P0154R1>`__","LWG","constexpr std::hardware_{constructive,destructive}_interference_size","Jacksonville","|Partial| [#note-P0154]_","15.0"
+"`P0154R1 <https://wg21.link/P0154R1>`__","LWG","constexpr std::hardware_{constructive,destructive}_interference_size","Jacksonville","|Complete| [#note-P0154]_","19.0"
```
Comment (h-vetinari) [inline libcxx/docs/Status/Cxx17Papers.csv:38]: Now that we've reinstated the footnote, we _may_ want to keep the 15.0 as the version when this was implemented? 🤔

Comment (mordante) [reply libcxx/docs/Status/Cxx17Papers.csv:38]: I prefer to keep 19, 15 only works with GCC. The combination of GCC and libc++ is not too common.

**Pull request merged**